docs/marketplace/dev/front-end/ui-components-library/data-transformers/collate/data-configurators/index.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Data Transformer Data Configurators
+description: This document provides details about the Data Transformer Data Configurators service in the Components Library.
+template: concept-topic-template
+---
+
+
+This document provides details about the Data Transformer Data Configurators service in the Components Library.
+
+## Overview
+
+Data Transformer Data Configurators is an Angular Service that implements re-population of data based on configuration.
+This allows backend systems to control where the re-population data.
+
+Data Transformer Data Configurators are used in the Datasource service.
+
+```html
+<spy-table
+  [config]="{
+    datasource: {
+      ...                                                   
+      transform: {
+        type: 'collate',
+        configurator: {
+          type: 'table',
+        },
+        ...  
+      },
+    },
+  }"
+></spy-table>
+```
+
+## Main Service
+
+The main module provides an opportunity to register a configurator by key via static method `withConfigurators`. It assigns the object of configurators to the `DataTransformerConfiguratorTypesToken` under the hood.
+
+The main service injects all registered types from the `DataTransformerConfiguratorTypesToken` and `DataTransformerConfigurator`.
+
+Resolve method finds specific service from the `DataTransformerConfiguratorTypesToken` by `type` (from the argument) and returns observable with data by `DataTransformerConfigurator.resolve`.
+
+## Data Transformer Data Configurator
+
+Data Transformer Data Configurator is basically an Angular Service that encapsulates the algorithm of how the data is configured.
+
+Data Transformer Data Configurator must implement a specific interface (DataTransformerConfigurator) and then be registered to the Root Module via `CollateDataTransformerModule.withConfigurators()`.
+
+```ts
+///// Module augmentation
+declare module '@spryker/data-transformer.collate' {
+    interface DataTransformerConfiguratorRegistry {
+        custom: CustomDataTransformerConfiguratorService;
+    }
+}
+
+//// Services implementation
+@Injectable({ providedIn: 'root' })
+export class CustomDataTransformerConfiguratorService implements DataTransformerConfigurator {
```

Comment:
Please, add `import { DataTransformerConfiguratorConfigT } from '@spryker/data-transformer.collate', that explicitly will be understandable where we can get `DataTransformerConfiguratorConfigT`